### PR TITLE
feat: [CDS-37920]: Return URL to pipeline/workflow that a trigger kicked off

### PR DESCRIPTION
**Pull request by vcalasansh**
## Describe your changes

This PR adds a field "uiSetupUrl" to the response of deployments triggered with `curl`. This field contains the UI URL to access the settings of the pipeline or workflow that was triggered:

- In case the execution type is *Workflow*, the URL is like `#/account/{account_id}/app/{app_id}/workflows/{workflow_id}/details`
- In case the execution type is *Pipeline*, the URL is like `#/account/{account_id}/app/{app_id}/pipelines/{pipeline_id}/edit`

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/35605)

### CODE REVIEW

**Review by Fernando-Dourado — APPROVED** [review]
> Reviewed 2 of 3 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35605)* status: :shipit: complete! ✓

File: 400-rest/src/main/java/software/wings/service/impl/WebHookServiceImpl.java [hunk 1]
```diff
@@ -137,6 +137,14 @@ private String getUiUrl(
     }
   }
 
+  private String getUiSetupUrl(boolean isPipeline, String accountId, String appId, String workflowId) {
+    if (isPipeline) {
+      return format("%s#/account/%s/app/%s/pipelines/%s/edit", getBaseUrlUI(), accountId, appId, workflowId);
```
Comment (lasdox) [inline 400-rest/src/main/java/software/wings/service/impl/WebHookServiceImpl.java:142]: i think this is not the exact url, please look at:
software.wings.graphql.datafetcher.execution.PipelineExecutionController#EXECUTION_URL_TEMPLATE

File: 400-rest/src/main/java/software/wings/service/impl/WebHookServiceImpl.java [hunk 2]
```diff
@@ -702,13 +710,15 @@ private void validateEventType(WebHookTriggerCondition triggerCondition, Map<Str
...
   @VisibleForTesting
   public Response constructSuccessResponse(String appId, String accountId, WorkflowExecution workflowExecution) {
-    WebHookResponse webHookResponse = WebHookResponse.builder()
-                                          .requestId(workflowExecution.getUuid())
-                                          .status(workflowExecution.getStatus().name())
-                                          .apiUrl(getApiUrl(accountId, appId, workflowExecution.getUuid()))
-                                          .uiUrl(getUiUrl(PIPELINE == workflowExecution.getWorkflowType(), accountId,
-                                              appId, workflowExecution.getEnvId(), workflowExecution.getUuid()))
-                                          .build();
+    Boolean isPipeline = PIPELINE == workflowExecution.getWorkflowType();
```
Comment (lasdox) [inline 400-rest/src/main/java/software/wings/service/impl/WebHookServiceImpl.java:713]: I think this isPipeline variable is not necessary here, you can just pass the workflowType and parse it in if else block or switch statement

Comment (vcalasansh) [reply 400-rest/src/main/java/software/wings/service/impl/WebHookServiceImpl.java]: Thank you for the suggestion, fixed it.

**Review by lasdox — APPROVED** [review]
> Reviewed 2 of 3 files at r1, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35605)* status: 1. unresolved discussions.

**Review by vcalasansh — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35605)* status: 2. unresolved discussions. 

___
*[`400-rest/src/main/java/software/wings/service/impl/WebHookServiceImpl.java` line 142 at r2](https://reviewable.io/reviews/harness/harness-core/35605#-N7WiUGaGuPMGcJZYKzW-r2-142:-N7phqBc6K6IUDt3rx0-:brztodx) ([raw file](https://github.com/harness/harness-core/blob/f4b545817c2f165e3dc76b3f6f51d04fafc24bc1/400-rest/src/main/java/software/wings/service/impl/WebHookServiceImpl.java#L142)):*
<details><summary><i>Previously, lasdox (Lucas Sales) wrote…</i></summary><blockquote>

i think this is not the correct url, please look at:
software.wings.graphql.datafetcher.execution.PipelineExecutionController#EXECUTION_URL_TEMPLATE
</blockquote></details>

After discussion, we concluded that the url is correct.

**Review by lasdox — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35605)* status: :shipit: complete! ✓

**Pull request merged**